pkg/kube/client.go

```diff
@@ -596,13 +615,32 @@ func (c *Client) waitForResources(timeout time.Duration, created Result) error {
 				if err != nil {
 					return false, err
 				}
+
 				for _, r := range rs.Items {
-					list, err := getPods(client, value.Namespace, r.Spec.Selector.MatchLabels)
+					currentReplicaSet, err := client.ReplicaSets(value.Namespace).Get(r.Name)
```

Comment:
Does this `Get` have to happen? They should be up to date from the Replica sets query above. You should just be able to just use `rs.Items`